runtime/service/src/test/java/org/apache/polaris/service/events/jsonEventListener/aws/cloudwatch/AwsCloudWatchEventListenerTest.java

```diff
@@ -0,0 +1,432 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.polaris.core.PolarisCallContext;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.service.events.AfterTableRefreshedEvent;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.CreateLogGroupRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.CreateLogStreamRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeLogGroupsRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeLogGroupsResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeLogStreamsRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DescribeLogStreamsResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsResponse;
+
+class AwsCloudWatchEventListenerTest {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AwsCloudWatchEventListenerTest.class);
+
+  private static final LocalStackContainer localStack =
+      new LocalStackContainer(
+              containerSpecHelper("localstack", AwsCloudWatchEventListenerTest.class)
+                  .dockerImageName(null))
+          .withServices(LocalStackContainer.Service.CLOUDWATCHLOGS);
+
+  private static final String LOG_GROUP = "test-log-group";
+  private static final String LOG_STREAM = "test-log-stream";
+  private static final String REALM = "test-realm";
+  private static final String TEST_USER = "test-user";
+  private static final Clock clock = Clock.systemUTC();
+
+  @Mock private AwsCloudWatchConfiguration config;
+
+  private ExecutorService executorService;
+  private AutoCloseable mockitoContext;
+
+  enum TestMode {
+    LOCALSTACK,
+    MOCKED
```

Comment:
For give me if this was from a previous review but I'm not sure I understand why "mocked" mode is useful here. It makes the test code very complicated and I'm not sure it tests very much. The LocalStack impl seems to be a much better test.

Reply:
It was asked from a previous [review](https://github.com/apache/polaris/pull/1965#discussion_r2210737076). I think the main thing it does is give a way to test the code in a mocked fashion, which will be similar to other future CSP product implementations, as they do not have a LocalStack equivalence.

Reply:
Talked offline with @eric-maynard - he is fine with removing the "mocked"-mode tests. Will reflect in the next revision.